src/System.Web.OData/OData/Builder/Conventions/Attributes/ColumnAttributeEdmPropertyConvention.cs

```diff
@@ -43,6 +45,11 @@ public ColumnAttributeEdmPropertyConvention()
             }
 
             var columnAttribute = attribute as ColumnAttribute;
+            if (columnAttribute != null && columnAttribute.Order > 0)
+            {
+                ODataQueryOptions.ColumnOrder.Add(edmProperty.Name, columnAttribute.Order);
```

Comment:
One of the reasons I asked for a second model in the test was to deal with potential name collisions.  You have avoided that by creating models with different order by attribute names but I'm still worried this code can't handle the case where two attributes of the same name (from two different entitles in a model OR from two different models) have different ordering. How does your code account for those cases?